### PR TITLE
Enhancement/use gson

### PR DESCRIPTION
**Pull request by przybylski**
* Reduce the complexity of hand parsing nodes, use gson
* Add a wrapper to avoid parsing server response

### CODE REVIEW

**Comment by AndyScherzinger** [comment]
:+1: Woohoo, nice one @przybylski

File: src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java [hunk 1]
```diff
@@ -140,73 +124,25 @@ protected RemoteOperationResult run(OwnCloudClient client) {
...
-                JSONObject respOCS = respJSON.getJSONObject(NODE_OCS);
-                JSONObject respData = respOCS.getJSONObject(NODE_DATA);
+                Gson gson = new Gson();
+                JsonParser parser = new JsonParser();
+                JsonObject respJson = (JsonObject)parser.parse(response);
+                Type respType = new TypeToken<ServerResponse<UserInfo>>(){}.getType();
+                ServerResponse<UserInfo> ocsResponse = gson.fromJson(respJson, respType);
 
-                UserInfo userInfo = new UserInfo();
+                UserInfo userInfo = ocsResponse.ocs.data;
```
Comment (mario) [inline src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java]: Woot! What about using getters instead? getOcs().getData()? :)

Comment (przybylski) [reply src/com/owncloud/android/lib/resources/users/GetRemoteUserInfoOperation.java]: Both ways for me. not using setters mimics traversing JSON object

**Comment by mario** [comment]
👍 

@przybylski ready for merging?

**Comment by przybylski** [comment]
Yes and no. Its complete in terms of implementation. If we want to generalise it and use it on other classes then no.

**Comment by mario** [comment]
As far as I'm concerned, we should use it all over the place.

**Comment by przybylski** [comment]
I made few changes as a generalisation, have a look and let me know if we want to go that way.

**Comment by mario** [comment]
@przybylski very good, yup. I went similar way with Talk (but used Retrofit & LoganSquare).

**Comment by przybylski** [comment]
If we are ok with incrementally change all other operations (which I can do) than it is redy to review.

**Comment by mario** [comment]
Yup, sure. 👍 
@AndyScherzinger can you ok one more time pls?

**Comment by AndyScherzinger** [comment]
👍 sure :)

**Pull request merged**